### PR TITLE
fix/content-type-int

### LINKED ISSUE #234: Waku v2 contentTopic type
Implement the spec contentType change: https://github.com/vacp2p/specs/pull/217

### PR DESCRIPTION
**Pull request by decanus**
closes #234

### CODE REVIEW

File: examples/v2/chat2.nim [hunk 1]
```diff
@@ -31,7 +31,7 @@ const Help = """
 """
 
 const DefaultTopic = "waku"
-const DefaultContentTopic = "dingpu"
+const DefaultContentTopic = ContentTopic(1)
```
Comment (oskarth) [inline examples/v2/chat2.nim:34]: Can we make the constructor us "dingpu" in byte format? Instead of using 1 everywhere.

Comment (decanus) [reply examples/v2/chat2.nim:34]: @oskarth yeah, was going to do that. Was just testing to see if everything works.

**Comment by decanus** [comment]
Tested with Chat2

File: examples/v2/chat2.nim [hunk 2]
```diff
@@ -31,7 +31,9 @@ const Help = """
 """
 
 const DefaultTopic = "waku"
-const DefaultContentTopic = "dingpu"
+
+const Dingpu = "dingpu".toBytes
+const DefaultContentTopic = ContentTopic(uint32.fromBytes([Dingpu[0], Dingpu[1], Dingpu[2], Dingpu[3]]))
```
Comment (oskarth) [inline examples/v2/chat2.nim:36]: This seems weird and manual to me. Why not do something like `cast[seq[byte](...)`?

Comment (decanus) [reply examples/v2/chat2.nim:36]: This currently seems like the most decent way, the fromBytes function needs a fixed sized array that is the length of the int.

Comment (kdeme) [reply examples/v2/chat2.nim:36]: You should be able to pass just `Dingpu`, as long as it contains 4 bytes that should work. Or you can take the 0..4 slice to be explicit I guess.

**Review by oskarth — APPROVED** [review]
lgtm

**Review by jm-clius — APPROVED** [review]
Will also merge these into my Unsubscribe-from-contentTopic changes.

**Pull request merged**